packages/compiler/src/server/tspconfig/completion.ts

```diff
@@ -14,24 +33,48 @@ export async function provideTspconfigCompletionItems(
   tspConfigPosition: Position,
   context: {
     fileService: FileService;
-    emitterProvider: EmitterProvider;
+    compilerHost: CompilerHost;
+    emitterProvider: LibraryProvider;
+    linterProvider: LibraryProvider;
     log: (log: ServerLog) => void;
   },
 ): Promise<CompletionItem[]> {
-  const { fileService, emitterProvider, log } = context;
+  const { fileService, compilerHost, emitterProvider, linterProvider, log } = context;
   const target = resolveYamlScalarTarget(tspConfigDoc, tspConfigPosition, log);
   if (target === undefined) {
     return [];
   }
-  const items = resolveTspConfigCompleteItems(await fileService.getPath(tspConfigDoc), target);
+
+  // Variable interpolation
+  const variableInterpolationItems = resolveVariableInterpolationCompleteItems(
+    target.yamlDoc,
+    target.path,
+    tspConfigDoc.getText().slice(target.nodePostionRange.pos, target.curPos),
+  );
+  if (variableInterpolationItems.length > 0) {
+    return variableInterpolationItems;
+  }
+
+  const items = resolveTspConfigCompleteItems(
+    await fileService.getPath(tspConfigDoc),
+    target,
+    tspConfigPosition,
```

Comment:
do you still need this? we have it in target, don't we?

Reply:
Yes, because it's relative to the full text in target, but relative to the current line in tspConfigPosition